lib/workers/evm_server.rb

```diff
@@ -62,6 +62,8 @@ def kill_servers
   end
 
   def refresh_servers_to_monitor
+    @servers_to_monitor = servers_from_db if servers_to_monitor.empty?
```

Comment:
Note, this is initialized in the `initialize` above on line 13.  That could occur "before" servers are created such as when you do the initial deployment which calls `EvmServer.new.start` which initializes this object, sets the `@servers_to_monitor` to `[]` because no servers are created and the in `start`, it calls `EvmDatabase.seed_primordial`, which creates the server.  We need refetch the servers from the db to avoid starting the local server twice from `start_servers` once in `refresh_servers_to_monitor` and once in the `as_each_server  { start_server }`

Reply:
I wonder if this line should move into start_servers instead of here, as this is also called during other methods as well.  It seems start_servers is the "exception", so perhaps this line makes more sense there?

Reply:
Good point.  I actually made a `servers_to_monitor` getter method originally and made that update the ivar if it was empty.  I can see putting this in `start` also.  Let me look at it again.  If you're unsure of where this should be fixed too, it's probably a good idea to better understand what's happening.  I was trying to avoid making large changes but maybe one of these locations will make the most sense without too much change.

Reply:
Honestly, I want to make this change because I think it's very weird that we have two methods that start servers, currently `refresh_servers_to_monitor` and  `as_each_server { start_server }` in `start_servers`.

I think we can rename `refresh_servers_to_monitor` and let it deal with desired state of servers but that makes all the tests fails so I'm reluctant to make it even though it feels like the right step to take.



```diff
diff --git a/lib/workers/evm_server.rb b/lib/workers/evm_server.rb
index f7356496ff..09872db00a 100644
--- a/lib/workers/evm_server.rb
+++ b/lib/workers/evm_server.rb
@@ -10,7 +10,7 @@ class EvmServer

   def initialize
     $log ||= Rails.logger
-    @servers_to_monitor = servers_from_db
+    @servers_to_monitor = []
   end

   def start
@@ -39,7 +39,6 @@ class EvmServer

   def start_servers
     refresh_servers_to_monitor
-    as_each_server { start_server }
   end

   def monitor_servers
```